src/components/input.js

```diff
@@ -199,114 +199,64 @@ export const ValidatedInput = props => {
-      renderSuggestionsContainer: ({ containerProps, children }) => {
-        return children && h(AutocompleteSuggestions, { containerProps, children, target: id })
-      },
-      renderSuggestion: v => v,
-      renderInputComponent: inputProps => h(TextInput, { ...props, ...inputProps, style, type: 'search', nativeOnChange: true }),
-      theme: {
-        container: { width: '100%' },
-        suggestionsList: { margin: 0, padding: 0 },
-        suggestion: styles.suggestion,
-        suggestionHighlighted: { backgroundColor: colors.light(0.4) }
-      }
-    })
-  }
-}
-
-export class AutocompleteSearch extends Component {
-  static propTypes = {
-    value: PropTypes.string.isRequired,
-    onChange: PropTypes.func.isRequired,
-    onSuggestionSelected: PropTypes.func.isRequired,
-    suggestions: PropTypes.array,
-    renderSuggestion: PropTypes.func,
-    theme: PropTypes.object
-  }
-
-  static defaultProps = {
-    suggestions: [],
-    renderSuggestion: _.identity
-  }
-
-  constructor(props) {
-    super(props)
-    this.state = { show: false }
-    this.id = props.id || _.uniqueId('AutocompleteSearch_')
-  }
-
-  render() {
-    const { value, onChange, onSuggestionSelected, suggestions, renderSuggestion, theme, ...props } = this.props
-    const { show } = this.state
-    return h(Autosuggest, {
-      id: this.id,
-      inputProps: { id: this.id, value, onChange: onChange ? (e => onChange(e.target.value)) : undefined, ...props },
-      suggestions: show ? (value ? [value, ..._.filter(Utils.textMatch(value), suggestions)] : suggestions) : [],
-      onSuggestionsFetchRequested: () => this.setState({ show: true }),
-      onSuggestionsClearRequested: () => this.setState({ show: false }),
-      onSuggestionSelected: (e, { suggestionValue }) => onSuggestionSelected(suggestionValue),
-      getSuggestionValue: _.identity,
-      shouldRenderSuggestions: value => value.trim().length > 0,
-      renderSuggestionsContainer: ({ containerProps, children }) => {
-        return children && h(AutocompleteSuggestions, { containerProps, children, target: this.id })
-      },
-      renderSuggestion,
-      renderInputComponent: inputProps => h(TextInput, { nativeOnChange: true, ...inputProps }),
-      theme: _.merge({
-        container: { width: '100%' },
-        suggestionsList: { margin: 0, padding: 0 },
-        suggestion: styles.suggestion,
-        suggestionHighlighted: { backgroundColor: colors.light(0.4) }
-      }, theme)
-    })
-  }
+  return h(Downshift, {
+    selectedItem: value,
+    onInputValueChange: onChange,
+    inputId: id
+  }, [
+    ({ getInputProps, getMenuProps, getItemProps, isOpen, openMenu, toggleMenu, highlightedIndex }) => {
+      return div({ style: { width: '100%' } }, [
+        h(TextInput, getInputProps({
+          style,
+          type: 'search',
+          onFocus: openOnFocus ? openMenu : undefined,
+          onKeyDown: e => {
+            if (e.key === 'Escape') {
```

Comment:
What is the intention of this logic with 'Escape'? It seems like it is trying to have 'Escape' escape the dropdown if that is open without escaping the whole model. However this is not working for me in testing (if my understanding is wrong then no issue there) However after I choose an option from the dropdown 'Escape' no longer does anything unless I re-click/focus on the model which I don't think is intended?

Reply:
Hmm, this seems to be working for me.
The escape behavior should be this: 
1. Input unfocused: no change from normal, modal will close
2. Input focused:
  a. Menu open: menu will close
  b. Text entered: text is cleared
  c. Neither: menu will show; however normal behavior won't be prevented, so modal will close

Reply:
Hmm testing today yielded the behavior you outlined above, I am not sure what I was running into last week, apologies, all looks good!